students/WesleyM/Lesson4/trigrams.py

```diff
@@ -0,0 +1,42 @@
+import re
+import random
+
+def open_file(string):
+    '''Opens, reads and closes a .txt file'''
+    f = open(string)
+    secret_data = f.read()
+    f.close()
+    return secret_data
+
+def ngrams(input, n):
+    '''Takes text and makes n-grams'''
+    input = input.split(' ')
+    output = []
+    for i in range(len(input)-n+1):
+        output.append(input[i:i+n])
+    return output[:-1]
+
+def main():
+    dict = {}
+    text = (re.sub('[^A-Za-z]+', ' ', "{}".format(open_file('sherlock_small.txt')))).lower()
+    trigram = ngrams(text,3)
+
+    for i in range(len(trigram)):
+        if dict.get(trigram[i][0] + ' ' + trigram[i][1], False) == False:
+            dict[trigram[i][0] + ' ' + trigram[i][1]] = [trigram[i][2]]
+        else:
+            dict[trigram[i][0] + ' ' + trigram[i][1]].append(trigram[i][2])
```

Comment:
Check out `dict.setdefault()` or `defaultdict` from the `collections` module of the Python standard library for a way to simplify these kinds of look ups and assignments